src/drivers/osd/msp_osd/uorb_to_msp.cpp

```diff
@@ -279,6 +339,54 @@ msp_raw_gps_t construct_RAW_GPS(const sensor_gps_s &vehicle_gps_position,
 	return raw_gps;
 }
 
+msp_rendor_latitude_t construct_rendor_GPS_LAT(const sensor_gps_s &vehicle_gps_position)
+{
+	msp_rendor_latitude_t lat;
+
+	lat.screenYPosition = 0x0A; //
+	lat.screenXPosition = 0x29; //
+
+	if (vehicle_gps_position.fix_type >= 2) {
+		sprintf(&lat.str[0], "%.6f", vehicle_gps_position.latitude_deg);
+
+	} else {
+		sprintf(&lat.str[0], "%.6f", 0.0);
+	}
+
+	return lat;
+}
+
+msp_rendor_longitude_t construct_rendor_GPS_LON(const sensor_gps_s &vehicle_gps_position)
+{
+	msp_rendor_longitude_t lon;
+
+	lon.screenYPosition = 0x09; //
+	lon.screenXPosition = 0x29; //
+
+	if (vehicle_gps_position.fix_type >= 2) {
+		sprintf(&lon.str[0], "%.6f", vehicle_gps_position.longitude_deg);
+
+	} else {
+		sprintf(&lon.str[0], "%.6f", -0.0);
+	}
+
+	return lon;
+}
+
+msp_rendor_satellites_used_t construct_rendor_GPS_NUM(const sensor_gps_s &vehicle_gps_position)
+{
+	msp_rendor_satellites_used_t num;
+
+	num.screenYPosition = 0x08; //
+	num.screenXPosition = 0x29; //
+	char num_str[4];
+	sprintf(num_str, "%d", vehicle_gps_position.satellites_used % 100); // max 99
```

Comment:
```suggestion
	sprintf(num_str, "%d", vehicle_gps_position.satellites_used); 
```
if we actually want to do the check for >99 I would add logic before to constrain it. I don't think it's needed though, since this will never happen

Reply:
Modified